dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/communityManager/SingleDataRequestsTest.kt

```diff
@@ -113,33 +123,32 @@ class SingleDataRequestsTest {
             dataType = SingleDataRequest.DataType.lksg,
             reportingPeriods = setOf("2022"),
         )
-        val storedDataRequest = requestControllerApi.postSingleDataRequest(singleDataRequest).first()
-        val storedDataRequestId = UUID.fromString(storedDataRequest.dataRequestId)
-
-        val retrievedDataRequest = requestControllerApi.getDataRequestById(storedDataRequestId)
-
-        assertEquals(companyIdOfNewCompany, retrievedDataRequest.datalandCompanyId)
-        assertEquals(RequestStatus.open, retrievedDataRequest.requestStatus)
-    }
-
-    @Test
-    fun `post a single data request with a PermId that matches a Dataland company and assert the correct matching`() {
-        val validPermId = System.currentTimeMillis().toString()
-        val companyIdOfNewCompany = getIdForUploadedCompanyWithIdentifiers(permId = validPermId)
-        val storedDataRequest = postStandardSingleDataRequest(validPermId)
-        val storedDataRequestId = UUID.fromString(storedDataRequest.dataRequestId)
-
-        val retrievedDataRequest = requestControllerApi.getDataRequestById(storedDataRequestId)
-
-        assertEquals(companyIdOfNewCompany, retrievedDataRequest.datalandCompanyId)
-        assertEquals(RequestStatus.open, retrievedDataRequest.requestStatus)
+        val timestampBeforeSingleRequest = retrieveTimeAndWaitOneMillisecond()
+        val response = requestControllerApi.postSingleDataRequest(singleDataRequest)
+        checkThatAllReportingPeriodsAreTreatedAsExpected(
+            singleDataRequestResponse = response,
+            expectedNumberOfStoredReportingPeriods = 1,
+            expectedNumberOfDuplicateReportingPeriods = 0,
+        )
+        val newlyStoredRequests = getNewlyStoredRequestsAfterTimestamp(timestampBeforeSingleRequest)
+        checkThatTheAmountOfNewlyStoredRequestsIsAsExpected(newlyStoredRequests, 1)
+        assertEquals(
+            companyIdOfNewCompany,
+            newlyStoredRequests[0].datalandCompanyId,
+            "The company ID of the newly stored data request does not match the expected one.",
+        )
+        assertEquals(
```

Comment:
This test can technically be merged with the first test to reduce testing time. But keeping them separate is also fine